recipes/opencv/3.x/conanfile.py

```diff
@@ -0,0 +1,395 @@
+            del self.options.fPIC
+        if self.settings.os != "Linux":
+            del self.options.with_gtk
+
+    def configure(self):
+        if self.settings.compiler == "Visual Studio" and \
+           "MT" in str(self.settings.compiler.runtime) and self.options.shared:
+            raise ConanInvalidConfiguration("Visual Studio and Runtime MT is not supported for shared library.")
+        if self.settings.compiler == "clang" and tools.Version(self.settings.compiler.version) < "4":
+            raise ConanInvalidConfiguration("Clang 3.x can build OpenCV 3.x due an internal bug.")
+        if self.settings.compiler.cppstd and self.options.with_openexr:
+            tools.check_min_cppstd(self, 11)
+        if self.options.shared:
+            del self.options.fPIC
+        self.options["libtiff"].jpeg = self.options.with_jpeg
+        self.options["jasper"].with_libjpeg = self.options.with_jpeg
+
+    def requirements(self):
+        self.requires("zlib/1.2.11")
+        if self.options.with_jpeg == "libjpeg":
+            self.requires("libjpeg/9d")
+        elif self.options.with_jpeg == "libjpeg-turbo":
+            self.requires("libjpeg-turbo/2.0.5")
+        if self.options.with_png:
+            self.requires("libpng/1.6.37")
+        if self.options.with_jasper:
+            self.requires("jasper/2.0.16")
+        if self.options.with_openexr:
+            self.requires("openexr/2.5.2")
+        if self.options.with_tiff:
+            self.requires("libtiff/4.1.0")
+        if self.options.with_eigen:
+            self.requires("eigen/3.3.8")
+        if self.options.parallel == "tbb":
+            self.requires("tbb/2020.2")
+        if self.options.with_webp:
+            self.requires("libwebp/1.1.0")
+        if self.options.contrib:
+            self.requires("freetype/2.10.4")
+            self.requires("harfbuzz/2.7.2")
+            self.requires("gflags/2.2.2")
+            self.requires("glog/0.4.0")
+        if self.options.get_safe("with_gtk"):
+            self.requires("gtk/system")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version][0])
+        os.rename("opencv-{}".format(self.version), self._source_subfolder)
+
+        tools.get(**self.conan_data["sources"][self.version][1])
+        os.rename("opencv_contrib-{}".format(self.version), self._contrib_folder)
+
+    def _patch_opencv(self):
+        tools.rmdir(os.path.join(self._source_subfolder, "3rdparty"))
+        if self.options.with_openexr:
+            find_openexr = os.path.join(self._source_subfolder, "cmake", "OpenCVFindOpenEXR.cmake")
+            tools.replace_in_file(find_openexr, "SET(OPENEXR_LIBSEARCH_SUFFIXES x64/Release x64 x64/Debug)", "")
+            tools.replace_in_file(find_openexr, "SET(OPENEXR_LIBSEARCH_SUFFIXES Win32/Release Win32 Win32/Debug)", "")
+            tools.replace_in_file(find_openexr, r'SET(OPENEXR_ROOT "C:/Deploy" CACHE STRING "Path to the OpenEXR \"Deploy\" folder")', "")
```

Comment:
I think this pach can be removed.
It is a cache variable, so when passing `OPENEXR_ROOT` by argument to cmake, this `set` is overridden.

Reply:
Indeed! Removed!